python/ray/serve/_private/version.py

```diff
@@ -124,7 +124,14 @@ def _get_serialized_options(
         should prompt a deployment version update.
         """
         reconfigure_dict = {}
-        for option_name, field in self.deployment_config.__fields__.items():
+        # TODO(aguo): Once we only support pydantic 2, we can remove this if check.
+        # In pydantic 2.0, `__fields__` has been renamed to `model_fields`.
```

Comment:
Q: any public APIs to access fields?

Reply:
in pydantic 2, it is now `model_fields` and public. I don't think there was a public api in pydantic v1. Although I'm not sure if it was even private in pydantic v1. There were examples of using it in the docs.